scripts/build-environments-tree.js

```diff
@@ -0,0 +1,86 @@
+var RE = {
+  engine: /^(\w+?)(tp|[\d_]+)?(?:corejs[23])?$/,
+  version: /^(\d{4})(\d{2})(\d{2})$|^(\d+)(?:_(\d+))?(?:_(\d+))?$/
+};
+
+module.exports = buildEnvsTree;
+
+// This function returns an object mapping "browser id" -> "parent browser id"
+function buildEnvsTree(data) {
+  var result = {};
+  // Pre-initialize the "result" object to make it have the same order as "data"
+  Object.keys(data).forEach(function (id) { result[id] = null; });
+
+  var envs = Object.create(null);
+
+  Object.keys(data).forEach(function (id) {
+    var parsed = parseEnvId(id);
+    if (!envs[parsed.name]) envs[parsed.name] = [];
+    envs[parsed.name].push(parsed);
```

Comment:
it seems that `result[id] = null;` from line 12 could be moved here to avoid second iteration over `data` object keys

Reply:
Actually, I think I can just remove it since the code is not relying on the ordering of `result` (it did in a previous attempt).